envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -222,6 +222,17 @@ message HttpConnectionManager {
   // downstream. If this is false or not set, Envoy will instead strip the
   // "Expect: 100-continue" header, and send a "100 Continue" response itself.
   bool proxy_100_continue = 18;
+
+  // If
+  // :ref:`use_remote_address
+  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.use_remote_address>`
+  // is true and remote_address_ipv4_mapped is true and the remote address is an IPv4 address, the
+  // address will be mapped to IPv6 before it is appended. This is useful for testing compatibility
+  // of upstream services that parse the header value. For example, 50.0.0.1 is represented as
+  // ::FFFF:50.0.0.1. See `IPv4-Mapped IPv6 Addresses
+  // <https://tools.ietf.org/html/rfc4291#section-2.5.5.2>`_ for details. This will also affect the
+  // :ref:`config_http_conn_man_headers_x-envoy-external-address` header.
+  bool remote_address_ipv4_mapped_ipv6 = 19;
```

Comment:
Should this be a verb to match other fields, e.g. `append_ipv6_mapped_remote_ipv4` or something similar?

Reply:
well we're already appending either way so i thought that would be confusing. this doesn't really control whether or not we append, it's a question of format. i thought for awhile about a good name and this was the best i could come up with. still not ideal and happy to take other suggestions.